### ISSUE TITLE
PgHero cleanup (dupe 1)

### ISSUE THREAD

**Issue by jennb33** [department-of-veterans-affairs/va.gov-team#87373]
This ticket is a dupe of ticket [80648](https://app.zenhub.com/workspaces/platform-product-team-633af4074573d06c3cda142a/issues/gh/department-of-veterans-affairs/va.gov-team/80648) (GH link: https://github.com/department-of-veterans-affairs/va.gov-team/issues/80648) for Sprint 5. Please refer to original ticket for comments. 

## Issue Description
PgHero (http://pghero-prod.vfs.va.gov/) is reporting some slow queries and, occasionally a long-running query.
**Updated screenshot 7/8/24**
<img width="254" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/10993987/1cf26194-beed-42f9-b282-fb664633fdee">

Some of the slowest and/or most frequently called queries should be addressed. We will want to report significant findings to the teams that own the related code. 

## Tasks
- [x] investigate the yellow alerts in PgHero
- [x] Fix issues or pass along findings to relevant teams
    - [x] Link conversations, tickets, or PRs as comments in this issue.

### Itemized tasks
Taken from the comments of https://github.com/department-of-veterans-affairs/va.gov-team/issues/80648
- [x] `index_ivc_champva_forms_on_form_uuid` Invalid index
- [x] De-duplicate Indexes. Create a migration to change the table to drop the index for the dupes. 
    - [x] On accreditations               
`index_accreditations_on_accredited_individual_id (accredited_individual_id)` is covered by `index_accreditations_on_indi_and_org_ids (accredited_individual_id, accredited_organization_id)`
    - [x] On async_transactions `index_async_transactions_on_transaction_id (transaction_id)`
is covered by `index_async_transactions_on_transaction_id_and_source (transaction_id, source)`
    - [x] On va_notify_in_progress_reminders_sent `index_va_notify_in_progress_reminders_sent_on_user_account_id (user_account_id)` is covered by  `index_in_progress_reminders_sent_user_account_form_id (user_account_id, form_id)`
- [x] Address the remaining slow queries (some work was done in https://github.com/department-of-veterans-affairs/vets-api/pull/17377)
- [x] Address high number of connections. 

## Success Metrics
The info in PgHero has been analyzed and acted upon.

## Acceptance Criteria
- [ ] Teams have been notified
- [ ] links to slack threads or tickets created from this are linked in the comments of this issue
- [ ] PG Lockout is created for extended run queries
---

## Validation
_Assignee to add steps to this section. List the actions that need to be taken to confirm this issue is complete. Include any necessary links or context. State the expected outcome._

**Comment by rmtolmach** [comment]
- Invalid index fix: https://github.com/department-of-veterans-affairs/vets-api/pull/17447
- Duplicate index fix: https://github.com/department-of-veterans-affairs/vets-api/pull/17451

**Comment by rmtolmach** [comment]
I merged the duplicate index PR this morning and dev and staging synced and pghero in dev and staging both say "No duplicate indexes" now 🎉 

I will watch sandbox and prod at the daily deploy to validate these.

**Comment by rmtolmach** [comment]
✅ prod and sandbox synced. This is the updated pghero screenshot:
<img width="254" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/10993987/34e68739-0ea3-4321-a79d-0955b96b5811">

**Comment by rmtolmach** [comment]
**High number of connections**: The default max connections for a `db.r5.xlarge` instance is 2000, so even though it's showing up as yellow in PgHero, it's not a concern. We have a monitor, [Platform - AWS RDS Connections Utilization](https://vagov.ddog-gov.com/monitors/100559), that will tell us if we're close to our limit. Ryan J. updated the monitor last week, and I'm making another pass of it in this PR: https://github.com/department-of-veterans-affairs/devops/pull/14530. Once this PR is merged, we can consider this PgHero issue addressed.

**Comment by jennb33** [comment]
7/11 update: @rmtolmach and Lindsey met, this is ready for a BE review then DevOps will need to do a terraform for it.

**Comment by jennb33** [comment]
7/15 update: Rebecca is going to send the PRs to @Kshitiz-devops for approval for the first bullet. The 1st bullet point needs to be reviewed by DevOps COP

**Comment by rmtolmach** [comment]
High number of connections is almost all done!

The manifest PRs were merged.

Updated PgHero screenshot for Prod:
![image](https://github.com/user-attachments/assets/903ab612-1ebc-467e-a002-8f0e50512cd1)

Only thing left to do is get the terraform applied (monitors updated) on https://github.com/department-of-veterans-affairs/devops/pull/14530 and that will happen tomorrow (tuesday). The slow queries issue was turned into another ticket: https://github.com/department-of-veterans-affairs/va.gov-team/issues/88315 meaning we can close this ticket after the monitors are terraformed tomorrow.

**Comment by rmtolmach** [comment]
Chris ran the terraform. Closing!

**Comment by LindseySaari** [comment]
## Validation 

- [x] [PgHero prod](http://pghero-prod.vfs.va.gov/) is green
- [x] The three updated monitors (see devops pr) look good.
- [x] Tickets created for follow-up work

Looks good!
